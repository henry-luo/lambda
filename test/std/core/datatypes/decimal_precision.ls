// Test: Decimal Precision and Rounding
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test decimal precision
let precise = 0.1m + 0.2m
precise
let number = 1.23456789m
number
let round_down = number.round(2, "down")
round_down
let round_up = number.round(2, "up")
round_up
let round_half_even = number.round(2)
round_half_even
let a = 1.0000000000000000000000000001m
a
let b = 1.0m
b
let exact = a - b
exact
let scaled = 1.2345m.set_scale(2)
scaled
let scaled_up = 1.2m.set_scale(4)
scaled_up
let div = 1.0m / 3.0m
div
1
