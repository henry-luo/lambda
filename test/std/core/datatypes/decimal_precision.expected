0.3
1.23456789
1.23
1.24
1.23
1.0000000000000000000000000001
1.0
0.0000000000000000000000000001
1.23
1.2000
0.3333333333333333333333333333
1
